tests/Agent/IntegrationTests/IntegrationTests/WCF/WCFServiceTests.cs

```diff
@@ -361,7 +362,14 @@ public override void DistributedTracing_SpanEvents()
             var unexpectedMethodNames = new List<string>();
 
             var includeASPPipelineEvents = _hostingModelOption == HostingModel.IIS && _aspCompatibilityOption == ASPCompatibilityMode.Enabled;
-            const int COUNT_ASP_PIPELINE_EVENTS = 10;
+
+            // In debug we consistently get 10 extra ASP pipeline events, in Release we consistently get 9.
```

Comment:
This is great and if it gets us a consistently passing integration test I'm all for it.  However, I think @tehbio had an alternative solution in mind where we make `PreExecuteRequestHandler` do something non-trivial in the test application so that it always runs regardless of build mode.  Thoughts?

Reply:
Spoke with folks offline, we're okay with the way this PR handles this debug/release difference for this test.